pallets/ajuna-tournament/src/lib.rs

```diff
@@ -0,0 +1,571 @@
+			ensure!(reward_table_total_dist <= 100, Error::<T, I>::InvalidTournamentConfig);
+
+			ensure!(
+				config.max_players > 0 && config.max_players <= MAX_PLAYERS,
+				Error::<T, I>::InvalidTournamentConfig
+			);
+
+			Ok(reward_table_total_dist)
+		}
+
+		fn try_get_current_tournament_id_for(
+			season_id: &T::SeasonId,
+		) -> Result<TournamentId, DispatchError> {
+			ActiveTournaments::<T, I>::get(season_id)
+				.ok_or_else(|| Error::<T, I>::NoActiveTournamentForSeason.into())
+		}
+
+		fn process_rank_fee_deposit(
+			payer: &AccountIdFor<T>,
+			fee_percentage: u8,
+		) -> Result<BalanceOf<T, I>, DispatchError> {
+			let base_deposit = T::RankDeposit::get();
+			let fee_deposit = base_deposit
+				.saturating_mul(fee_percentage.into())
+				.checked_div(&100_u32.into())
+				.unwrap_or_default();
+			let global_treasury_account = Self::global_treasury_account_id();
+
+			T::Currency::transfer(
+				payer,
+				&global_treasury_account,
+				fee_deposit,
+				ExistenceRequirement::KeepAlive,
+			)?;
+
+			Ok(base_deposit.saturating_sub(fee_deposit))
+		}
+
+		fn process_max_reward_payout(
+			payer: &AccountIdFor<T>,
+			tournament_treasury_account: &AccountIdFor<T>,
+			rank_deposit: BalanceOf<T, I>,
+			max_reward: BalanceOf<T, I>,
+		) -> DispatchResult {
+			let tournament_treasury_balance =
+				T::Currency::free_balance(tournament_treasury_account);
+			let tournament_deposit_left = max_reward.saturating_sub(tournament_treasury_balance);
+			let effective_tournament_deposit =
+				sp_std::cmp::min(tournament_deposit_left, rank_deposit);
+
+			if effective_tournament_deposit > 0_u32.into() {
+				T::Currency::transfer(
+					payer,
+					tournament_treasury_account,
+					effective_tournament_deposit,
+					ExistenceRequirement::KeepAlive,
+				)?;
+			}
+
+			if effective_tournament_deposit < rank_deposit {
+				let effective_global_deposit =
+					rank_deposit.saturating_sub(effective_tournament_deposit);
+				let global_treasury_account = Self::global_treasury_account_id();
+
+				T::Currency::transfer(
+					payer,
+					&global_treasury_account,
+					effective_global_deposit,
+					ExistenceRequirement::KeepAlive,
+				)?;
+			}
+
+			Ok(())
+		}
+
+		fn process_category_payout(
+			treasury_account: &AccountIdFor<T>,
+			_category: &T::RankCategory,
+			tournament_config: &TournamentConfigFor<T, I>,
+			rank_table: &PlayerTableFor<T, I>,
+		) -> DispatchResult {
+			let total_payout = T::Currency::free_balance(treasury_account);
+
+			for ((account, _), payout_perc) in rank_table
+				.iter()
+				.zip(&tournament_config.reward_table)
+				.filter(|(_, perc)| **perc > 0)
+			{
+				let account_payout = total_payout
+					.saturating_mul((*payout_perc).into())
+					.checked_div(&100_u32.into())
+					.unwrap_or_default();
+
+				if account_payout > 0_u32.into() {
+					T::Currency::transfer(
+						treasury_account,
+						account,
+						account_payout,
+						ExistenceRequirement::AllowDeath,
+					)?;
+				}
```

Comment:
Is it ok, if this fails midst payout? It will interrupt this function, and also the `try_finish_tournament_function`.

Reply:
I think this needs a proper meeting to discuss what do we want to do about it.